<h2>Soal 1: Membuat Objek DataFrames</h2>

- Buatlah objek dataframes dari objek dictionary, Bebas!

In [ ]:
# Membuat dataframes dari dictionary
import pandas as pd

data = {
    'Nama': ['Muhammad', 'Naufal', 'Rafi'],
    'Usia': [20,21,22],
    'Kota': ['Depok', 'Jakarta', 'Bandung'],
    'Status': ['Single','Menikah', 'Menikah']
}

df_data = pd.DataFrame(data)
df_data

Nama  Usia     Kota   Status
0  Muhammad    20    Depok   Single
1    Naufal    21  Jakarta  Menikah
2      Rafi    22  Bandung  Menikah

<h2>Soal 2: Membaca data dari file</h2>

Download file bernama [diamonds.csv](https://drive.google.com/uc?export=download&id=1qQiM9utQUThTPq8jk8stPMh_y2zfqYOd), kemudian baca data tersebut dan assign ke dalam suatu variabel kemudian print.

In [ ]:
# Membaca data dan memasukannya kedalam suatu variabel

diamonds_1 = pd.read_csv('diamonds-1.csv')
diamonds_1

Unnamed: 0  carat        cut color clarity  depth  table  price     x  \
0               1   0.23      Ideal     E     SI2   61.5   55.0    326  3.95   
1               2   0.21    Premium     E     SI1   59.8   61.0    326  3.89   
2               3   0.23       Good     E     VS1   56.9   65.0    327  4.05   
3               4   0.29    Premium     I     VS2   62.4   58.0    334  4.20   
4               5   0.31       Good     J     SI2   63.3   58.0    335  4.34   
...           ...    ...        ...   ...     ...    ...    ...    ...   ...   
53935       53936   0.72      Ideal     D     SI1   60.8   57.0   2757  5.75   
53936       53937   0.72       Good     D     SI1   63.1   55.0   2757  5.69   
53937       53938   0.70  Very Good     D     SI1   62.8   60.0   2757  5.66   
53938       53939   0.86    Premium     H     SI2   61.0   58.0   2757  6.15   
53939       53940   0.75      Ideal     D     SI2   62.2   55.0   2757  5.83   

          y     z  
0      3.98  2.43  
1      3.84  2.31  
2      4.07  2.31  
3      4.23  2.63  
4      4.35  2.75  
...     ...   ...  
53935  5.76  3.50  
53936  5.75  3.61  
53937  5.68  3.56  
53938  6.12  3.74  
53939  5.87  3.64  

[53940 rows x 11 columns]

Expected Result:

![alt text](https://drive.google.com/uc?id=1BvgM5mJEVfj0fllrjkBfQwn0QIFkry_t)

<h2>Soal 3: DataFrame vs Series</h2>

- Apa perbedaan Series dengan list dan dictionary
- Apa perbedaan dataframe dan series
- Buatlah suatu data series

Jawab: 
* Series berasal dari library pandas, dan pada series tipe data yang disimpan harus seragam berbeda dengan list dan dictionary yang dapat menyimpan tipe data yang berbeda.

* Pada dataframe terdapat dua index yaitu index baris dan index kolom. dataframe memiliki 2 objek dimensi sedangkan series memiliki 1 objek dimensi.

*

In [ ]:
mata_uang = pd.Series(["Indonesia", "Jepang", "Amerika"], index = ['IDR', 'JPY', 'USD'])
mata_uang

IDR    Indonesia
JPY       Jepang
USD      Amerika
dtype: object

<h2>Soal 4: Pemeriksaan Data Sederhana</h2>

Di soal no 2 kamu telah membaca data tentang diamonds. pada soal ini cobalah suatu metode dari pandas untuk mengetahui beberapa data pertama dan beberapa data terakhir. kemudian sebutkan nilai dari:
- baris pertama dari column price
- baris terakhir dari column color

tunjukan cara kalian menemukan nilai tersebut di block di bawah ini.

In [ ]:
print('baris pertama kolom price:', diamonds_1['price'].iloc[0])
print('baris terakhir kolom color:', diamonds_1['color'].iloc[-1]);

baris pertama kolom price: 326
baris terakhir kolom color: D


<h2>Soal 5: Deskriptif Statistik</h2>

Jelaskan apa itu deskriptis statistik!

Jawab: 

Deskriptif statistik adalah sebuah metode untuk memberikan informasi dengan cara mengumpulkan data lalu menyajikan nya dalam suatu himpunan data [1]. Dengan adanya deskriptif statistik data-data tersebut akan tersaji ringkas dan rapih. deskriptif statistik terdiri dari pemusatan data, penyebaran data, serta kecendrungan dari suatu gugus data [2].

[1] Ronald E.Walpole. Pengantar Statistika, halaman 2-5". 1993. Jakarta: PT Gramedia Pustaka Utama.

[2] Dergibson Siagian & Sugiarto. Metode Statistika untuk Bisnis dan Ekonomi, halaman 4-6". 2002. Jakarta: PT Gramedia Pustaka Utama.

<h2>Soal 6: Practice Deskriptif Statistik</h2>

- Gunakan suatu metode dari pandas untuk mengetahui deskriptif statistik dari suatu data
- Berapa nilai rata-rata dari column price
- Berapa nilai standar deviasi dari column depth
- Berapa nilai maximum dari column carat

In [ ]:
print('nilai rata rata kolom price:', diamonds_1['price'].mean())
print('nilai standar deviasi:', diamonds_1['depth'].std())
print('nilai maksimum dari carat:', diamonds_1['carat'].max())

nilai rata rata kolom price: 3932.799721913237
nilai standar deviasi: 1.4326213188337733
nilai maksimum dari carat: 5.01


Jawab:

* nilai rata rata kolom price: 3932.799721913237
* nilai standar deviasi: 1.4326213188337733
* nilai maksimum dari carat: 5.01